src/main/java/seedu/address/model/country/CountryManager.java

```diff
@@ -67,22 +74,10 @@ public boolean hasCountryNote(Country country, Note countryNote) {
      * @param countryNote The country note to be added.
      */
     public void addCountryNote(Country country, Note countryNote) {
-        if (isValidCountryCode(country.getCountryCode())) {
-            countryCodeMap.get(country.getCountryCode()).addCountryNote(countryNote);
+        if (!countryNotesMap.containsKey(country)) {
+            return;
         }
```

Comment:
This should be an assert / exception instead, since our countryNotesMap already contains all possible countries, and if there is a country not contained in our map, it should not have been created in the first place